transport/http2_server.go

```diff
@@ -137,7 +142,11 @@ func newHTTP2Server(conn net.Conn, config *ServerConfig) (_ ServerTransport, err
 	if config.ReadBufferSize > 0 {
 		readBufSize = config.ReadBufferSize
 	}
-	framer := newFramer(conn, writeBufSize, readBufSize)
+	maxHeaderListSize := defaultServerMaxHeaderListSize
+	if config.MaxHeaderListSize != nil {
+		maxHeaderListSize = *config.MaxHeaderListSize
+	}
+	framer := newFramer(conn, writeBufSize, readBufSize, maxHeaderListSize)
```

Comment:
Only relevant to the server side: Should `Framer.MaxHeaderListSize` be set only after fist settings' ack is received?
The other side might start with an unlimited MaxHeaderListSize(https://tools.ietf.org/html/rfc7540#section-6.5.2) and start sending header frames before our settings frame is received by it.

Reply:
Offline discussion decides that at current stage, enforcement after ack will not be necessary and add complexity to the code.